hazelcast/src/main/java/com/hazelcast/spi/impl/executionservice/impl/SchedulingExecutorServiceDelegate.java

```diff
@@ -27,12 +29,14 @@
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.TimeoutException;
 
-final class ScheduledExecutorServiceDelegate implements ScheduledExecutorService {
+import static com.hazelcast.util.Preconditions.checkNotNull;
+
+public final class SchedulingExecutorServiceDelegate implements SchedulingExecutorService {
```

Comment:
well, it delegates to a given executor service. this is significant implementation aspect, the reason why we have it at all. otherwise we might very well use just the `ScheduledExecutorService` from JDK. 

what about `DelegatingSchedulingExecutorService` ?